billing-service/src/main/java/org/egov/demand/service/BillServicev2.java

```diff
@@ -284,26 +288,48 @@ public BillResponseV2 generateBill(GenerateBillCriteria billCriteria, RequestInf
 				.tenantId(billCriteria.getTenantId())
 				.email(billCriteria.getEmail())
 				.consumerCode(consumerCodes)
-				.isPaymentCompleted(false)
 				.receiptRequired(false)
 				.demandId(demandIds)
 				.build();
 
 		/* Fetching demands for the given bill search criteria */
-		List<Demand> demands = demandService.getDemands(demandCriteria, requestInfo);
+		List<Demand> demandsWithMultipleActive = demandService.getDemands(demandCriteria, requestInfo);
+
+		if (demandsWithMultipleActive.isEmpty()) {
+			throw new CustomException(EG_BS_BILL_NO_DEMANDS_FOUND_KEY, EG_BS_BILL_NO_DEMANDS_FOUND_MSG);
+		}
+
+		//filter the demands which are fully paid
+		demandsWithMultipleActive = demandsWithMultipleActive.stream().filter(demand -> !demand.getIsPaymentCompleted()).collect(Collectors.toList());
+
+		List<Demand> demands = filterMultipleActiveDemands(demandsWithMultipleActive);
 
 		List<BillV2> bills;
 
 		if (!demands.isEmpty())
 			bills = prepareBill(demands, requestInfo);
 		else
-			throw new CustomException(EG_BS_BILL_NO_DEMANDS_FOUND_KEY, EG_BS_BILL_NO_DEMANDS_FOUND_MSG);
+			return getBillResponse(Collections.emptyList());
 
 		BillRequestV2 billRequest = BillRequestV2.builder().bills(bills).requestInfo(requestInfo).build();
 		//kafkaTemplate.send(notifTopicName, null, billRequest);
 		return create(billRequest);
 	}
 
+	private List<Demand> filterMultipleActiveDemands(List<Demand> demands) {
```

Comment:
I fail to see the purpose of this method? Please explain

Reply:
The name for the method might be different , this is just sort the demands.

Reply:
The name for the method might be different , this is just sort to the demands.

Reply:
sorting is not needed.